boards/arm/lora_e5_dev_board/board.cmake

```diff
@@ -5,7 +5,7 @@ board_runner_args(pyocd "--flash-opt=-O reset_type=hw")
 board_runner_args(pyocd "--flash-opt=-O connect_mode=under-reset")
 board_runner_args(jlink "--device=STM32WLE5JC" "--speed=4000" "--reset-after-load")
 board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")
-board_runner_args(blackmagicprobe "--gdb-serial=/dev/ttyBmpGdb" "--connect-srst")
+board_runner_args(blackmagicprobe "--gdb-serial=/dev/ttyBmpGdb" "--connect-rst")
```

Comment:
Unrelated to this PR but why is this forcing a specific serial port name through a udev rule?

Reply:
ttyBmpGdb is the the alias used in the official documentation and udev rules (https://github.com/blackmagic-debug/blackmagic/blob/main/driver/99-blackmagic-plugdev.rules) it's used by a few boards, ideally the runner would try to see if that exists automatically and then guess something else, that could be a valid followup.

Reply:
It is very limited, you can only have one programmer connected to a PC at any one time for example and requires that users have installed udev rules, which I can't see linked in the documentation. It also completely breaks support for working on windows

Reply:
I've opened an RFC on this issue: https://github.com/zephyrproject-rtos/zephyr/issues/53859

Reply:
Yeah well, right now the fallback is just `/dev/ttyACM0` so unless I missed something windows (or mac) support and multiple programmers are already broken. But yeah agreed that the runner should have some smartness and these per board overrides should go, thanks for opening the RFC.

Reply:
Added a couple patches to address the flag as well. Broadening the scope of the PR but since it hasn't been reviewed yet I've figured I'll just expand it.